src/commands/deploy/showCoreToolsWarning.ts

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { DialogResponses, IActionContext, nonNullValue, openUrl } from "@microsoft/vscode-azext-utils";
+import { MessageItem } from "vscode";
+import { getLocalFuncCoreToolsVersion } from "../../funcCoreTools/getLocalFuncCoreToolsVersion";
+import { FuncVersion, funcVersionLink, tryParseFuncVersion } from "../../FuncVersion";
+import { localize } from "../../localize";
+import { getWorkspaceSetting, updateGlobalSetting } from "../../vsCodeConfig/settings";
+
+export async function showCoreToolsWarning(context: IActionContext, projectVersion: FuncVersion): Promise<void> {
+    const showCoreToolsWarningKey: string = 'showCoreToolsWarning';
+    if (getWorkspaceSetting<boolean>(showCoreToolsWarningKey)) {
+        const coreToolsVersion = await getLocalFuncCoreToolsVersion(context, undefined)
+        if (coreToolsVersion !== null) {
+            const localVersion = nonNullValue(tryParseFuncVersion(coreToolsVersion), 'localCoreToolsVersion');
+            let result: MessageItem | undefined;
+
+            if (localVersion === FuncVersion.v2 || localVersion === FuncVersion.v3) {
+                result = await showCoreToolsEOLWarning(context, localVersion);
+                while (result === DialogResponses.learnMore) {
+                    await openUrl(funcVersionLink);
+                    result = await showCoreToolsEOLWarning(context, localVersion);
+                }
+                if (result === DialogResponses.dontWarnAgain) {
+                    await updateGlobalSetting(showCoreToolsWarningKey, false);
+                }
+            } else if (localVersion !== projectVersion) {
+                result = await showCoreToolsMismatchWarning(context, localVersion, projectVersion);
+                while (result === DialogResponses.learnMore) {
```

Comment:
You should be able to put this logic in the `showCoreTools...Warning` methods. I think that'll clean up this conditional a little bit and make it a bit easier to read :)